shared/fs/files-loading-hoc.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import {compose, connect, lifecycle, setDisplayName, type Dispatch, type TypedState} from '../util/container'
+import * as FsGen from '../actions/fs-gen'
+import * as Types from '../constants/types/fs'
+import * as Constants from '../constants/fs'
+
+const mapStateToProps = (state: TypedState) => ({
+  syncingPaths: state.fs.uploads.syncingPaths,
+})
+
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  loadFolderList: (path: Types.Path) => dispatch(FsGen.createFolderListLoad({path})),
+  loadFavorites: () => dispatch(FsGen.createFavoritesLoad()),
+})
+
+const mergeProps = ({syncingPaths}, {loadFolderList, loadFavorites}, {routeProps, routePath}) => {
+  const path = routeProps.get('path', Constants.defaultPath)
+  return {
+    syncingPaths,
+    loadFolderList,
+    loadFavorites,
+    path,
+    routePath,
+  }
+}
+
+const FilesLoadingHoc = compose(
```

Comment:
just fyi we prefer to just make a simple wrapper class vs using lifecycle. lifecycle loses a lot of flow typing